### PR TITLE
Jetpack Boost: Improve error handling and response consistency

### PR DESCRIPTION
**Pull request by pyronaur**
## Proposed changes:
- Refactor DataSyncError and related DataSyncHooks to handle error states more consistently and provide utility methods like isAborted.
- Implement response formatting functions in PHP action and endpoint classes for consistent error and success responses.
- Simplify page cache setup and error handling in the Jetpack Boost plugin by moving cache setup calls and removing redundant hooks.
- Standardize error handling across the plugin by using the DataSyncError.info method and standardize-error utility.

### Other information:
- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
N/A

## Testing instructions:
- Trigger error states in the DataSync client and verify that the 'isAborted' method correctly identifies aborted requests.
- Invoke actions in the PHP classes that result in errors or success and ensure that the response formats are consistent with the new methods.
- Enable and disable page cache in Jetpack Boost to test if the setup and error handling works without issues.
- Check various components that handle errors to confirm that the standardized error messages are displayed correctly.

### CODE REVIEW

**Comment by dilirity** [comment]
When enabling the cache, there's a double notification (one for setup complete and another for cache cleared).

Make sure you clear the cache before disabling i to cause this to happen.

https://github.com/Automattic/jetpack/assets/11799079/f9530793-821d-4a4f-8de9-3720d033c841

File: projects/plugins/boost/app/assets/src/js/lib/stores/page-cache.ts [hunk 1]
```diff
@@ -29,17 +34,35 @@ export function usePageCache() {
...
+		},
+		mutationOptions: {
+			onError: error => {
+				if ( error instanceof DataSyncError ) {
+					return setError( error.info() );
+				}
+				const standardizedError = standardizeError( error );
+				setError( {
+					code: 'unknown_error',
+					message: standardizedError.message || 'Unknown error occurred.',
```
Comment (dilirity) [inline projects/plugins/boost/app/assets/src/js/lib/stores/page-cache.ts]: `Unknown error occurred.` is not translatable.

**Review by dilirity — CHANGES_REQUESTED** [review]
Other than the double notice thing above, this looks pretty good!

Great work!

One small thing and we're good to go.

**Review by dilirity — APPROVED** [review]
Let's merge this for now, as there are things this PR will unblock. We'll review what I found above later.

:shipit:

**Comment by thingalon** [comment]
Cherry-picked to 3.1.0 in dfb201341b.

**Pull request merged**